crates/extricrate/src/lib.rs

```diff
@@ -364,24 +364,85 @@ pub mod dependencies {
 
     pub type ModuleDependencies = HashMap<ModuleName, Vec<ModuleName>>;
 
-    /// List the (circular?) dependencies of modules inside the given crate, based on the use statements.
+    /// List the dependencies of modules inside the given crate, including circular, based on the use statements.
     pub fn list_dependencies(use_statements: &UseStatementMap) -> ModuleDependencies {
-        todo!()
+        let mut module_dependencies: HashMap<ModuleName, Vec<ModuleName>> = HashMap::new();
```

Comment:
```suggestion
        let mut module_dependencies: HashMap<ModuleName, HashSet<ModuleName>> = HashMap::new();
```

Otherwise we'll get duplicates.

Reply:
I'll leave it up to you whether you want to update the interface (`ModuleDependencies`) or convert that into a `Vec` before returning.

Reply:
I like returning the `HashSet` more so it's clearer what the function will do from the signature